### DISCUSSION TITLE
[expo-notifications] Push notification content JSON data is always null

### DISCUSSION THREAD

**Discussion by doougui** [expo/expo#18624] [Archived: Questions] [answered]
Hello. We are trying to implement our own back-end for push notifications, but when sending notifications with either FCM or Apple's APN, the JSON data field, that is supposed to be our additional data that we will use to open the page in the React Native application, is always null.

![image](https://user-images.githubusercontent.com/44846329/184343543-a8d46805-9770-4c32-bc48-0198c90dbd5a.png)

The payload we are using for FCM is this:

```json
{
  "to": "xxxxxx",
  "priority": "normal",
  "data": {
      "experienceId": "@app/app",
      "scopeKey": "@app/app",
      "title": "Sample title",
      "body": "Sample body",
      "data": { // this will be used to open the screen in the app
          "name": "ACTION_NAME",
          "payload": {
              "id": "331720c4-ea3f-485b-bab9-bf99b325a7f7"
          }
      }
  }
}
```

We also have tried something like this:

```json
{
  "to": "/topics/news",
  "notification": {
    "title": "Breaking News",
    "body": "New news story available."
  },
  "data": {
    "story_id": "story_12345"
  }
}
```

But it doesn't work either. Any ideas or recommendations?

**Comment by keith-kurak** [comment]
When using Expo push notifications and testing via the [push notification testing tool](https://expo.dev/notifications), I am able see the `data` field on `notification.request.content`. However, that's not the same test case, as it sounds like you're sending via APNS/ FCM directly. If you are doing this, you'll want to double-check on the format with Apple/ Google. For instance, [here is the FCM format for sending a combined data/ message title and body notification](https://firebase.google.com/docs/cloud-messaging/concept-options#notification-messages-with-optional-data-payload), which looks a little different from what you have above.

**Reply by doougui** [reply]
It turns out we were using the wrong payload. The correct payload is this one:

```json
{
    "to": "xxx",
    "priority": "normal",
    "data": {
        "experienceId": "@app/app",
        "scopeKey": "@app/app",
        "title": "Testing",
        "message": "This is the message",
        "body": {
            "name": "OPEN_PERSON",
            "payload": {
                "id": "xxx"
            }
        }
    }
}
```

Thanks for your help anyway :)

**Comment by doougui** [comment, marked as answer]
Solution: https://github.com/expo/expo/discussions/18624#discussioncomment-3434256
